pkgs/os-specific/darwin/duti/default.nix

```diff
@@ -10,15 +16,14 @@ stdenv.mkDerivation rec {
     sha256 = "1pg4i6ghpib2gy1sqpml7dbnhr1vbr43fs2pqkd09i4w3nmgpic9";
   };
 
-  nativeBuildInputs = [autoreconfHook];
-  buildInputs = [ApplicationServices];
-  configureFlags = [
-    "--with-macosx-sdk=/homeless-shelter"
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = lib.optional stdenv.hostPlatform.isDarwin apple-sdk_11;
 
-    # needed to prevent duti from trying to guess our sdk
-    # NOTE: this is different than stdenv.hostPlatform.config!
-    "--host=x86_64-apple-darwin18"
-  ];
+  # needed to prevent duti from trying to guess our sdk
+  # NOTE: this is different than stdenv.hostPlatform.config!
+  configureFlags =
+    lib.optionals stdenv.hostPlatform.isAarch64 [ "--host=aarch64-apple-darwin18" ]
+    ++ lib.optionals stdenv.hostPlatform.isx86_64 [ "--host=x86_64-apple-darwin18" ];
```

Comment:
Looking at https://github.com/moretension/duti/blob/8b5b9a08702d5de6a5bce57fade593fca46e126e/aclocal.m4, we do actually still need to work around their outdated SDK checks, but we should do it a little differently.

I suggest:

```nix
  preConfigure = ''
    configureFlagsArray+=(
      "--with-macosx-sdk=$SDKROOT"
      "--with-macosx-deployment-target=$MACOSX_DEPLOYMENT_TARGET"
    )
  '';
```

This should also obviate the need for `--host`.

Reply:
Sure, changed. But I can’t see how to remove the host flag - you still end up with failure on newer systems as host will be set to an unhandled version (eg `darwin24` on latest macOS). Any suggestions, or are you happy with the latest commit?

Reply:
Hmm, that’s very unfortunate, it’s doing some strange and unnecessary stuff. I see two options:

* Pass `--host=${stdenv.hostPlatform.parsed.cpu.name}-apple-darwin20` and put `substituteInPlace Makefile.in --replace-fail '@macosx_arches@' ""` in a `postPatch` phase to work around its suboptimal architecture selection (might be able to get away without the latter part, not sure).

* Add a patch that comments out `DUTI_CHECK_SDK` and `DUTI_CHECK_DEPLOYMENT_TARGET` in `configure.ac` and replaces [all this](https://github.com/moretension/duti/blob/8b5b9a08702d5de6a5bce57fade593fca46e126e/Makefile.in#L17-L20) with just `OPTOPTS=	@OPTOPTS@`. Then we don’t need any configure flags at all and it should just work indefinitely. This is probably the cleaner approach.